indexing-service/src/main/java/io/druid/indexing/common/task/Tasks.java

```diff
@@ -74,7 +73,8 @@
           toBeAccumulated.add(interval);
         } else {
           compactIntervals.add(JodaUtils.umbrellaInterval(toBeAccumulated));
-          toBeAccumulated = Lists.newArrayList(interval);
+          toBeAccumulated.clear();
```

Comment:
It's hard to imagine that a method `umbrellaInterval()`, accepting an `Iterable` of `Interval`s and returning an `Interval`, a final immutable object (defined in a library) that stores just begin and end of the interval, to store or modify the `Iterable` argument. Also, as per "Effective Java", the responsibility for making defensive copies is on the method (API) implementation side, not on the caller side.

Reply:
Joda Interval objects are immutable so I think this point is moot. It doesn't matter if `JodaUtils.umbrellaInterval` returns a new Interval instance or not.

Reply:
Here is a simple example of `umbrellaInterval()` to break this method. Note that every caller currently passes a list. This method can be changed to accept a list without further change.

```java
  public static Interval umbrellaInterval(List<Interval> intervals)
  {
    if (intervals.size() == 1) {
      return intervals.get(0);
    }
...
```

> Also, as per "Effective Java", the responsibility for making defensive copies is on the method (API) implementation side, not on the caller side.

I basically agree on this. If you want to keep as it is, please add a javadoc to `umbrellaInterval()` to make sure that it always returns a new instance.

Reply:
@jihoonson but `umbrellaInterval()` is called before the list is cleared. The version that you presented will work fine as well, unless there is *concurrent* modification of the passed list, that is a total mess and would break `umbrellaInterval()` anyway.

Reply:
Oh, you're correct. This looks fine.